src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -3407,6 +3279,106 @@ public override bool HandleCall (MethodBody callingMethodBody, MethodReference c
+									} else {
+										reflectionContext.RecordUnrecognizedPattern ($"Expression call '{calledMethod.FullName}' inside '{callingMethodBody.Method.FullName}' was detected with 1st argument which cannot be analyzed");
+									}
+								}
+							}
+							break;
+
+						//
+						// static Field (Expression, Type, String)
+						// static Property (Expression, Type, String)
+						//
+						case var fieldOrProperty when ((fieldOrProperty == "Field" || fieldOrProperty == "Property")
+							&& calledMethod.DeclaringType.Name == "Expression"
+							&& calledMethod.Parameters.Count == 3
+							&& calledMethod.Parameters [1].ParameterType.FullName == "System.Type"): {
+
+								reflectionContext.AnalyzingPattern ();
+
+								foreach (var value in methodParams [1].UniqueValues ()) {
+									if (value is SystemTypeValue systemTypeValue) {
+										// systemTypeValue.TypeRepresented is `declaringType`
+										foreach (var stringParam in methodParams [2].UniqueValues ()) {
+											if (stringParam is KnownStringValue stringValue) {
+												bool staticOnly = methodParams [0].Kind == ValueNodeKind.Null;
+												if (calledMethod.Name [0] == 'P') {
+													MarkPropertiesFromReflectionCall (ref reflectionContext, systemTypeValue.TypeRepresented, stringValue.Contents, staticOnly);
+												} else {
+													MarkFieldsFromReflectionCall (ref reflectionContext, systemTypeValue.TypeRepresented, stringValue.Contents, staticOnly);
+												}
+											} else if (stringParam is MethodParameterValue) {
+												// TODO: Check if parameter is annotated.
+												reflectionContext.RecordUnrecognizedPattern ($"Expression call '{calledMethod.FullName}' inside '{callingMethodBody.Method.FullName}' was detected with 3rd argument which cannot be analyzed");
+											} else {
+												reflectionContext.RecordUnrecognizedPattern ($"Expression call '{calledMethod.FullName}' inside '{callingMethodBody.Method.FullName}' was detected with 3rd argument which cannot be analyzed");
+											}
+										}
+									} else if (value == NullValue.Instance) {
+										reflectionContext.RecordHandledPattern ();
+									} else if (value is MethodParameterValue) {
+										// TODO: Check if parameter is annotated.
+										reflectionContext.RecordUnrecognizedPattern ($"Expression call '{calledMethod.FullName}' inside '{callingMethodBody.Method.FullName}' was detected with 2nd argument which cannot be analyzed");
+									} else {
+										reflectionContext.RecordUnrecognizedPattern ($"Expression call '{calledMethod.FullName}' inside '{callingMethodBody.Method.FullName}' was detected with 2nd argument which cannot be analyzed");
+									}
+								}
+							}
+							break;
+
+						//
+						// static New (Type)
+						//
+						case "New" when calledMethod.DeclaringType.Name == "Expression"
```

Comment:
This can be annotated by placing `[DynamicallyAccessedMembers(DefaultConstructor)]` on the parameter - we don't need to special case in the linker.

Reply:
For now you can emulate that by calling something like this:
```C#
RequireDynamicallyAccessedMembers (ref reflectionContext, DynamicallyAccessedMemberKind.DefaultConstructor, methodParameters[0]);
```

That will do basically the same thing as adding the attribute would. And we can later add the attribute to the framework for real (which we should for all the intrinsic methods anyway).